sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetColumnVector.java

```diff
@@ -75,6 +75,11 @@ final class ParquetColumnVector {
     this.isPrimitive = column.isPrimitive();
 
     if (missingColumns.contains(column)) {
+      if (ParquetRowIndexUtil.isRowIndexColumn(column)) {
```

Comment:
How expensive is this call?

Reply:
It's a string comparison (so O(column name length), but in practice cheaper) that happens at most once per column per scan task. I believe it's in the same ballpark as all the other initialization happening at the beginning of the scan.